src/plot_api/plot_api.js

```diff
@@ -956,6 +956,12 @@ Plotly.addTraces = function addTraces(gd, traces, newIndices) {
     if(!Array.isArray(traces)) {
         traces = [traces];
     }
+
+    // make sure traces do not repeat existing ones
+    traces = traces.map(function(trace) {
+        return Lib.extendFlat({}, trace);
```

Comment:
@dfcreative Thanks for looking into :beers:  I believe this leads to the desired behaviour. We'll wait for @cpsievert to reply to https://github.com/plotly/plotly.js/issues/1083#issuecomment-260387606 to confirm though.

Do you know exactly why passing existing trace objects to `addTraces` cause an infinite loop?